js/initcontext.go

```diff
@@ -110,7 +119,47 @@ func newBoundInitContext(base *InitContext, ctxPtr *context.Context, rt *goja.Ru
 		programs:          programs,
 		compatibilityMode: base.compatibilityMode,
 		logger:            base.logger,
+
+		shares:     base.shares,
+		sharesLock: base.sharesLock,
+	}
+}
+
+// XShare is a constructor returning a shareable read-only array (general objects coming)
+// indentified by the name and having their contents be whatever the call returns - again only
+// arrays currently supported.
+func (i *InitContext) XShare(name string, call goja.Callable) goja.Value {
```

Comment:
As I mentioned in slack, I'd really like not to have new k6-specific superglobal names like `open()`, `__VU`, etc. Instead, this should be accessible via `k6/data/SharedArray` (so, `import {SharedArray} from "k6/data";`) or something like that. The `k6/data` package will probably be reused by https://github.com/loadimpact/k6/issues/1539 for other things like segmented iterators and maybe XML/JSON/etc. streaming parsers and segmentation (though those probably will be sub-packages)

Reply:
btw now it should be easy to move this outside of the `InitContext`, you can just have the shares registry in the `InitEnvironment` that was added in https://github.com/loadimpact/k6/pull/1706